pilot/pkg/networking/plugin/envoyfilter/envoyfilter_test.go

```diff
@@ -0,0 +1,158 @@
+			result:         true,
+		},
+		{
+			name:        "match outbound sidecar http listeners on 10.10.10.0/24:80, with port name prefix http-*",
+			inputParams: inputParams,
+			listenerIP:  net.ParseIP("10.10.10.10"),
+			direction:   OutboundDirection,
+			matchCondition: &networking.EnvoyFilter_ListenerMatch{
+				PortNumber:       80,
+				PortNamePrefix:   "http",
+				ListenerType:     networking.EnvoyFilter_ListenerMatch_SIDECAR_OUTBOUND,
+				ListenerProtocol: networking.EnvoyFilter_ListenerMatch_HTTP,
+				Address:          []string{"10.10.10.0/24"},
+			},
+			result: true,
+		},
+		{
+			name:        "does not match: outbound sidecar http listeners on 10.10.10.0/24:80, with port name prefix tcp-*",
+			inputParams: inputParams,
+			listenerIP:  net.ParseIP("10.10.10.10"),
+			direction:   OutboundDirection,
+			matchCondition: &networking.EnvoyFilter_ListenerMatch{
+				PortNumber:       80,
+				PortNamePrefix:   "tcp",
+				ListenerType:     networking.EnvoyFilter_ListenerMatch_SIDECAR_OUTBOUND,
+				ListenerProtocol: networking.EnvoyFilter_ListenerMatch_HTTP,
+				Address:          []string{"10.10.10.0/24"},
+			},
+			result: false,
+		},
+		{
+			name:        "does not match: inbound sidecar http listeners with port name prefix http-*",
+			inputParams: inputParams,
+			direction:   OutboundDirection,
+			matchCondition: &networking.EnvoyFilter_ListenerMatch{
+				PortNamePrefix:   "http",
+				ListenerType:     networking.EnvoyFilter_ListenerMatch_SIDECAR_INBOUND,
+				ListenerProtocol: networking.EnvoyFilter_ListenerMatch_HTTP,
+			},
+			result: false,
+		},
+		{
+			name:        "does not match: outbound gateway http listeners on 10.10.10.0/24:80, with port name prefix http-*",
+			inputParams: inputParams,
+			listenerIP:  net.ParseIP("10.10.10.10"),
+			direction:   OutboundDirection,
+			matchCondition: &networking.EnvoyFilter_ListenerMatch{
+				PortNumber:       80,
+				PortNamePrefix:   "http",
+				ListenerType:     networking.EnvoyFilter_ListenerMatch_GATEWAY,
+				ListenerProtocol: networking.EnvoyFilter_ListenerMatch_HTTP,
+				Address:          []string{"10.10.10.0/24"},
+			},
+			result: false,
+		},
+		{
+			name:        "does not match: outbound sidecar listeners on 172.16.0.1/16:80, with port name prefix http-*",
+			inputParams: inputParams,
+			listenerIP:  net.ParseIP("10.10.10.10"),
+			direction:   OutboundDirection,
+			matchCondition: &networking.EnvoyFilter_ListenerMatch{
+				PortNumber:       80,
+				PortNamePrefix:   "http",
+				ListenerType:     networking.EnvoyFilter_ListenerMatch_SIDECAR_OUTBOUND,
+				ListenerProtocol: networking.EnvoyFilter_ListenerMatch_HTTP,
+				Address:          []string{"172.16.0.1/16"},
+			},
+			result: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		ret := listenerMatch(tc.inputParams, tc.direction, tc.listenerIP, tc.matchCondition)
+		if tc.result != ret {
+			t.Errorf("%s: expecting %v but got %v", tc.name, tc.result, ret)
+		}
+	}
+}
```

Comment:
thanks - since no good work goes unpunished i'll also suggest adding tests for insertHTTPFilter and insertNetworkFilter since those have quite a few moving parts and are easily testable.

Reply:
Unit tests? I added integration tests for those.